### PR TITLE
add self update release channel

### PR DESCRIPTION
**Pull request by tarkah**
## Proposed Changes
  - Add release channel for self update (Release vs PreRelease)
  - PreRelease will show updates in Ajour for normal releases, as well as prerelease releases
  - Add picklist in settings to switch between channels

## Checklist

- [x] Tested on Windows
- [x] Tested on MacOS
- [ ] Tested on Linux
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

**Comment by tarkah** [comment]
We need to fix our self update logic to not break when triggered while Ajour is still running from an instance that was self updated. Since we will want to force a recheck of releases when channel is changed and a self update can now technically happen after one already did.

**Comment by tarkah** [comment]
@casperstorm I noticed when testing this that the "About" tab dissapears when an update is available.

**Comment by casperstorm** [comment]
> @casperstorm I noticed when testing this that the "About" tab dissapears when an update is available.

That's actually on purpose!
If it's unintuitive then we can look at changing it.

**Comment by tarkah** [comment]
> 
> 
> > @casperstorm I noticed when testing this that the "About" tab dissapears when an update is available.
> 
> That's actually on purpose!
> If it's unintuitive then we can look at changing it.

No I think it makes sense, I just never realized before so wanted to make sure it was intentional

File: crates/core/src/utility.rs [hunk 1]
```diff
@@ -87,6 +87,12 @@ pub async fn download_update_to_temp_file(
         .unwrap()
         .to_str()
         .unwrap()
+        // Trim `temp_` from the `current_exe` path in case this self update process
```
Comment (casperstorm) [inline crates/core/src/utility.rs:90]: Nice documented.

Comment (tarkah) [reply crates/core/src/utility.rs:90]: It's hard to make it make sense without way more context, but better than nothing :)

**Pull request merged**